src/phoenix/trace/llama_index/callback.py

```diff
@@ -248,45 +251,75 @@ def on_event_start(
     ) -> CBEventID:
         event_id = event_id or str(uuid4())
         event_data = self._event_id_to_event_data[event_id]
-        event_data["name"] = event_type.value
-        event_data["event_type"] = event_type
-        event_data["start_event"] = CBEvent(
+        event_data.name = event_type.value
+        event_data.event_type = event_type
+        event_data.start_event = CBEvent(
             event_type=event_type,
             payload=payload,
             id_=event_id,
         )
-        event_data["attributes"] = {}
         # Parse the payload to extract the parameters
         if payload is not None:
-            event_data["attributes"].update(
+            event_data.attributes.update(
                 payload_to_semantic_attributes(event_type, payload),
             )
 
         return event_id
 
-    @graceful_fallback(_null_fallback)
-    def on_event_end(
+    def _event_end_without_event_start_fallback(
         self,
         event_type: CBEventType,
         payload: Optional[Dict[str, Any]] = None,
         event_id: CBEventID = "",
         **kwargs: Any,
     ) -> None:
         event_data = self._event_id_to_event_data[event_id]
-        event_data.setdefault("name", event_type.value)
-        event_data.setdefault("event_type", event_type)
-        event_data["end_event"] = CBEvent(
+
+        # If the start event is missing, we create a dummy start event
+        event_data.start_event = CBEvent(
+            event_type=event_type,
+            id_=event_id,
+        )
+
+        event_data.name = event_type.value
+        event_data.event_type = event_type
+        event_data.end_event = CBEvent(
             event_type=event_type,
             payload=payload,
             id_=event_id,
         )
 
         # Parse the payload to extract the parameters
         if payload is not None:
-            event_data["attributes"].update(
+            event_data.attributes.update(
                 payload_to_semantic_attributes(event_type, payload, is_event_end=True),
             )
 
+    @graceful_fallback(_null_fallback)
+    @graceful_fallback(_event_end_without_event_start_fallback, (MissingEventStartError,))
```

Comment:
I think this is more complicated than necessary. We're just trying to solve a "missing value" problem by setting default values. It's not clear why it's necessary to go through an additional sequence of exception fallbacks, and what those extra steps are trying to accomplish.